### PR TITLE
Rework and clean up project/tooling files

### PR DESCRIPTION
**Pull request by jgarber623**
Hello, all! First, thanks so much for the hard work on this gem over the years.

Second, this PR is coming out of left field, so I apologize in advance if anything in the changeset runs afoul of the team's preferences. I'd love to get as much feedback as possible on this PR.

### Back Story

In looking through the open issues and PRs, I got the sense that @dissolve wasn't too happy with the reports coming out of [Code Climate](https://codeclimate.com/github/indieweb/microformats-ruby). I was a bit surprised, too, by the low scores on maintainability, etc.

So… that leads me into this PR. I think it would be wise to shore up some of the foundational tooling within the project before directly addressing the code-related things that Code Climate is flagging. Code Climate's a great collection of tools on its own, but it can sometimes be a bit over-zealous vs. integrating some of its tools directly into projects.

### The Proposed Changes

I've gone through and added fairly thorough commit comments that explain in detail my thinking behind many of the proposed changes. I'm happy to accept and respond to feedback/critique/etc. so let me know what you think.

**Edit:** Turns out commit comments aren't as useful in the context of a Pull Request as I might've thought. For clarity's sake, here's a list of links to the commits in my fork that properly display the comments I've added:

- jgarber623/microformats-ruby@bd01eca
- jgarber623/microformats-ruby@02b9c4e
- jgarber623/microformats-ruby@165bb9e
- jgarber623/microformats-ruby@685639f
- jgarber623/microformats-ruby@99be698
- jgarber623/microformats-ruby@cb65fe3
- jgarber623/microformats-ruby@610b8e6
- jgarber623/microformats-ruby@6778565
- jgarber623/microformats-ruby@12f23eb
- jgarber623/microformats-ruby@2e6c903
- jgarber623/microformats-ruby@cd54d0a
- jgarber623/microformats-ruby@060af18

☝️ Apologies for the weirdness!

### PR Goals

My main objective with this PR was to set the groundwork for future refactoring with the goal of improving Code Climate scores, test coverage, etc. There are very few things in this PR that amount to functionality changes or anything that would require cutting a release of the gem. This is all meta developer kind of stuff.

Thanks for your time and consideration and let me know what you think!

### CODE REVIEW

**Comment by veganstraightedge** [comment]
I left some questions in the comments. Overall, it seems good. But it’s @dissolve’s call. 👍🏻

**Comment by jgarber623** [comment]
@veganstraightedge Thanks for the review! I've addressed your questions and reverted a couple less-than-useful additions. Appreciate the questions and comments. 👍 

@dissolve Looking forward to seeing what you think when you have the time.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+1. Fix an existing [issue][issues] and submit a [pull request][pulls].
+1. Review open [pull requests][pulls].
+1. Report a new [issue][issues]. _Only do this after you've made sure the behavior or problem you're observing isn't already documented in an open issue._
+
+## Getting Started
+
+microformats-ruby is developed using Ruby 2.5.0 and is additionally tested against versions 2.2.9, 2.3.6, and 2.4.3 using [Travis CI](https://travis-ci.org/indieweb/microformats-ruby).
+
+Before making changes to microformats-ruby, you'll want to install Ruby 2.5.0. It's recommended that you use a Ruby version managment tool like [rbenv](https://github.com/rbenv/rbenv), [chruby](https://github.com/postmodern/chruby), or [rvm](https://github.com/rvm/rvm). Once you've installed Ruby 3.5.0 using your method of choice, install the project's gems by running:
```
Comment (dissolve) [inline CONTRIBUTING.md]: typos of Ruby 3.5.0

File: README.md [hunk 2]
```diff
@@ -1,243 +1,178 @@
-# Microformats (ruby)
+<img src="https://raw.githubusercontent.com/gist/jgarber623/5c4cb8712169f51b571d370084006699/raw/d5ece174e46d98770a4e66f313e81e37ee3a5bf0/microformats-ruby-logo.svg?sanitize=true" align="left" width="100">
```
Comment (dissolve) [inline README.md]: want to get confirmation from @tantek that it is okay to use the microformats logo with a recolor for a different project.  I can't see there being any complaint, but just want to check

File: CONTRIBUTING.md (hunk 1 above)
Comment (jgarber623) [reply CONTRIBUTING.md]: Ah, good catch! Thanks!

File: README.md (hunk 2 above)
Comment (jgarber623) [reply README.md]: Good call. The [microformats wiki's copyright page](http://microformats.org/wiki/copyright) has content dedicated to the public domain _after_ 2007-12-29. I believe the logo was added prior to that, so it'd be a good idea to get @tantek's feedback on this.

Easy enough to leave the logo image out if it's an issue.

Comment (jgarber623) [reply README.md]: Removed in 10d2fc0. Will open a separate issue for logo conversations.

**Pull request merged**